### PR TITLE
better sql editor

### PR DESCRIPTION
**Pull request by JulianWielga**
Resolves https://github.com/TouK/nussknacker/issues/1106

### CODE REVIEW

File: ui/client/components/graph/node-modal/editors/expression/SqlEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  return (
+    <RawEditor
+      onValueChange={valueChange}
+      expressionObj={value}
+      className={className}
+    />
+  )
+}
+
+const stringPattern = /(^'.*'$)|(^".*"$)/su
```
Comment (mproch) [inline ui/client/components/graph/node-modal/editors/expression/SqlEditor.tsx:49]: Let's add comment to regexp - what is its intent :)

File: ui/client/components/graph/node-modal/editors/expression/Formatter.ts [hunk 2]
```diff
@@ -17,18 +19,28 @@ export enum FormatterType {
...
-const valueStartsWithQuotationMark = (value) => startsWith(value, "\"") || startsWith(value, "\'")
-
-const quotationMark = value => valueStartsWithQuotationMark(value) ? valueQuotationMark(value) : defaultQuotationMark
-
 const stringSpelFormatter: Formatter = {
-  encode: value => quotationMark(value) + value + quotationMark(value),
-  decode: value => value.substring(1, value.length - 1),
+  encode: value => {
+    const qm = getQuotationMark(value)
+    return flow([
```
Comment (mproch) [inline ui/client/components/graph/node-modal/editors/expression/Formatter.ts:28]: how cool :+1:

File: ui/client/components/graph/node-modal/editors/expression/TemplatesUtils.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import {curry} from "lodash"
+
+const templatesSearch = /#\{\s*(.+?)\s*\}/gms
```
Comment (mproch) [inline ui/client/components/graph/node-modal/editors/expression/TemplatesUtils.ts:3]: Let's add comment to regexp - what is its intent.
And maybe some unit tests?

File: ui/client/components/graph/node-modal/editors/expression/QuotesUtils.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import {curry} from "lodash"
```
Comment (mproch) [inline ui/client/components/graph/node-modal/editors/expression/QuotesUtils.ts:1]: let's add comment/reflect in filename that these are spel quotation rules

File: ui/client/components/graph/node-modal/editors/expression/SqlEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      expressionObj={value}
+      className={className}
+    />
+  )
+}
+
+const quotedStringPattern = getQuotedStringPattern([QuotationMark.single, QuotationMark.double])
+
+const parseable = ({expression, language}: ExpressionObj) => {
+  return language === "spel" && quotedStringPattern.test(expression.trim())
```
Comment (lciolecki) [inline ui/client/components/graph/node-modal/editors/expression/SqlEditor.tsx:53]: I think we can do language as enum?

**Review by lciolecki — APPROVED** [review]
LGTM, but please wait for: https://github.com/TouK/nussknacker/pull/1456

File: ui/client/components/graph/node-modal/editors/expression/RawEditor.tsx [hunk 6]
```diff
@@ -6,7 +6,10 @@ import {Editor} from "./Editor"
 import {VariableTypes} from "../../../../../types"
 
 type Props = {
-  expressionObj: $TodoType,
+  expressionObj: {
+    expression: string,
+    language: string,
```
Comment (lciolecki) [inline ui/client/components/graph/node-modal/editors/expression/RawEditor.tsx:11]: Right now we can say it's `ExpressionLang`?

**Pull request merged**